test/Mono.Linker.Tests.Cases/Reflection/ActivatorCreateInstance.cs

```diff
@@ -532,6 +533,15 @@ private static void TestNullArgsOnKnownType ()
 			Activator.CreateInstance (type, BindingFlags.NonPublic | BindingFlags.Instance, null, null, CultureInfo.InvariantCulture);
 		}
 
+		[Kept]
+		[ExpectedNoWarnings]
+		private static void TestNullArgsNonPublicWithNonPublicAnnotation (
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicParameterlessConstructor | DynamicallyAccessedMemberTypes.NonPublicConstructors),
+			KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))] Type type)
+		{
+			Activator.CreateInstance (type, nonPublic: true);
```

Comment:
How does this test work? I assume we're trying to assert that the given type has its non-public constructors preserved, but how are we confirming that we're also not marking the other public constructors?

Wasn't the problem in the PR that we were marking extra things and getting additional warnings?

Reply:
Not really - the problem is that PublicParameterlessConstructor | NonPublicConstructors was an annotation which was not considered sufficient for calling CreateInstance(type, true). Which it should be.

Reply:
> Wasn't the problem in the PR that we were marking extra things and getting additional warnings?

I may have given you the wrong impression when we chatted - this additional warning was because we tried to _require_ (not mark) extra things on the input (which was an abstract DAMT value). But the extra requirements also would have resulted in extra things being marked for concrete input types (that just wasn't the failure mode we happened to hit).